core/src/main/scala/org/bitcoins/core/crypto/bip44/BIP44Coin.scala

```diff
@@ -0,0 +1,42 @@
+package org.bitcoins.core.crypto.bip44
+import org.bitcoins.core.crypto.bip32.{BIP32Child, BIP32Path}
+
+/**
+  * Represents a
+  * [[https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#change BIP44]]
+  * coin type.
+  */
+sealed abstract class BIP44Coin extends BIP32Path {
+  override def children: Vector[BIP32Child] =
+    Vector(BIP44Path.purposeChild, BIP32Child(toInt, hardened = true))
```

Comment:
I think we should have 

```scala
def node: Bip32Node = BIP44Node(toInt)
override def ancestors: Vector[BIP32Node] = Vector(BIP44Path.purposeChild)
override def path = ancectors.+=(node)
```

Reply:
and something like this for all other BIP44* implementations

Reply:
I don't really see the benefit of having `ancestors` in there. For it to be overridable it would have to be in the `BIP32Path` abstract class. But that's supposed to represent a complete path, and has no ancestors. It could be added to `BIP32Node`, but I don't really see the benefit there either